software/control_sw/src/cosmic_fengine.py

```diff
@@ -1506,7 +1534,14 @@ def delay_tracking(self):
                     continue
                 try:
                     # sleep till time to load == 0 (to allow for loading)
-                    time.sleep(self.phaserotate.get_time_to_load()/FPGA_CLOCK_RATE_HZ)  
+                    expected_sleep_duration = required_loadtime_s - time.time_ns()*1e-9
+                    feng_time_to_load = self.phaserotate.get_time_to_load()/FPGA_CLOCK_RATE_HZ
+                    assert (np.isclose(feng_time_to_load, expected_sleep_duration, atol=1e-4),
```

Comment:
This test is fine, but the critical test is that `time_to_load` _must be_ positive immediately after the load. If it's not, then what was intended to load will never stick

Reply:
Cool will add it.